libraries/abstractions/platform/freertos/iot_network_freertos.c

```diff
@@ -569,6 +573,49 @@ size_t IotNetworkAfr_Receive( void * pConnection,
 
 /*-----------------------------------------------------------*/
 
+size_t IotNetworkAfr_ReceiveUpto( void * pConnection,
+                                  uint8_t * pBuffer,
+                                  size_t bufferSize )
+{
+    int32_t socketStatus = 0;
+    size_t bytesReceived = 0;
+
+    /* Cast network connection to the correct type. */
+    _networkConnection_t * pNetworkConnection = ( _networkConnection_t * ) pConnection;
+
+    /* Caller should never pass a zero-length buffer. */
+    configASSERT( bufferSize > 0 );
+
+    /* Write the buffered byte. THIS IS A TEMPORARY WORKAROUND AND ASSUMES THIS
+     * FUNCTION IS ALWAYS CALLED FROM THE RECEIVE CALLBACK. */
+    if( pNetworkConnection->bufferedByteValid == true )
+    {
+        *pBuffer = pNetworkConnection->bufferedByte;
+        bytesReceived = 1;
+        pNetworkConnection->bufferedByteValid = false;
+    }
+
+    /* Block and wait for incoming data. */
+    socketStatus = SOCKETS_Recv( pNetworkConnection->socket,
+                                 pBuffer + bytesReceived,
+                                 bufferSize - bytesReceived,
+                                 0 );
+
+    if( socketStatus <= 0 )
+    {
+        IotLogError( "Error %ld while receiving data.", ( long int ) socketStatus );
+    }
+    else
+    {
+        IotLogDebug( "Successfully received %lu bytes.",
+                     ( unsigned long ) bytesReceived );
```

Comment:
It looks like if bytes were received by the SOCKETS_REcv call they are not included in this count or the count returned at the end of this function? Looks like it always returns 0 or one regardless how many bytes more than 1 you got? Is this the intended behavior? Does not look correct to me ...

Reply:
It is not the intended behavior. 
https://github.com/aws/amazon-freertos/pull/1078